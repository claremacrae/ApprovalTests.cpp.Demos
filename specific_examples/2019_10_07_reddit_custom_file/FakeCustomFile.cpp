#include "FakeCustomFile.h"

custom_file::custom_file(const std::string&)
{
    // Ignore filename for now - just use hard-coded values for testing
}

std::string custom_file::size() const
{
    return "100MB";
}
