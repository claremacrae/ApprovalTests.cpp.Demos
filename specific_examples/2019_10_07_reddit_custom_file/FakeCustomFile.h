#ifndef APPROVALTESTS_CPP_FAKECUSTOMFILE_H
#define APPROVALTESTS_CPP_FAKECUSTOMFILE_H

#include <string>

class custom_file
{
public:
    explicit custom_file(const std::string&);
    std::string size() const;
        
    int header1 = 42;
    std::string header2 = "FOO";
        
};

#endif //APPROVALTESTS_CPP_FAKECUSTOMFILE_H
