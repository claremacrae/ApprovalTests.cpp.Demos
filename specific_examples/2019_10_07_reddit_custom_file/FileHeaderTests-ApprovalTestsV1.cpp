#include "ApprovalTests.hpp"
#include "Catch.hpp"
#include "FakeCustomFile.h"

std::ostream &operator<<(std::ostream &os, const custom_file& test_file)
{
    os << "size:    " << test_file.size() << '\n';
    os << "header1: " << test_file.header1 << '\n';
    os << "header2: " << test_file.header2 << '\n';
    return os;
}

TEST_CASE( "header fields are correct - with ostream operator", "[header]" )
{
    using namespace ApprovalTests;
    custom_file big_file{"filename"};
    Approvals::verify(big_file);
}
