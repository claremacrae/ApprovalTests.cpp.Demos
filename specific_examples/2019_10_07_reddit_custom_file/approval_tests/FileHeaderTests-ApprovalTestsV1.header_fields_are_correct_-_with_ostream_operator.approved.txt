size:    100MB
header1: 42
header2: FOO

